test/cql-pytest/test_filtering.py

```diff
@@ -431,3 +431,25 @@ def test_token_bind_marker_name(cql, test_keyspace):
     with new_test_table(cql, test_keyspace, 'p1 int, p2 int, c1 int, c2 int, r int, PRIMARY KEY ((p1, p2), c1, c2)') as table:
         stmt = cql.prepare(f'SELECT * FROM {table} WHERE token(p1, p2) = ?')
         assert stmt.column_metadata[0].name == 'partition key token'
+
+
+# In the following test we check what error message appears if multi-column relation contains null values.    
+# Reproduces #13217
+def test_multi_column_relation_tuples_null_check(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, 'a int, b int, c int, d int, primary key (a, b, c, d)') as table:
+         with pytest.raises(InvalidRequest, match='Invalid null value in condition for column'): 
+                cql.execute(f'SELECT * FROM {table} WHERE a = 0 AND (b, c, d) IN ((null, 2, 3), (2, 1, 4))')
+         with pytest.raises(InvalidRequest, match='Invalid null value in condition for column'): 
+                cql.execute(f'SELECT * FROM {table} WHERE a = 0 AND (b, c, d) IN ((1, 2, 3), (2, 1, null))')      
+
+#This test shows how Scylla handles invalid number of items in one of the tuples of multi-column relation.
+#If the number of items is lesser than required, Scylla throws "Expected {} elements in value tuple, but got {}"
+#But if the number of items is greater than required, Scylla throws different error: 
+#"Invalid list literal for in((b,c,d)): value (1, 2, 3, 4) is not of type frozen<tuple<int, int, int>>"
+# Reproduces #13241, #13217
+def test_multi_column_relation_wrong_number_of_items_in_tuple(cql, test_keyspace):
```

Comment:
The error message "Invalid list literal" is correct. It's not beautiful that it's a different error message than the opposite case, but it's not really wrong, and it's not really a bug. Moreover, it doesn't make sense to say that a test reproduces an **open** issue (#13241) but the test passes!

So I'm wondering - could it be that your patch actually fixes #13241? Can you please check if the existing test for #13241 maybe started to pass too?

Reply:
My test doesn't fix the #13241 . A request with one tuple still causes Scylla to crash.  I shouldn't include "represents 13241" in this test. I think we should leave #13241 alone in the context of this issue.